WireUI/Sources/WireReusableUIComponents/PasswordTextField/PasswordField.swift

```diff
@@ -0,0 +1,195 @@
+
+    public init(
+        arePasswordRulesVisible: Bool = false,
+        isPasswordVisible: Bool = false,
+        password: String = "",
+        passwordValidator: any PasswordValidator
+    ) {
+        self.arePasswordRulesVisible = arePasswordRulesVisible
+
+        self.isPasswordVisible = isPasswordVisible
+        self.password = password
+        self.passwordValidator = passwordValidator
+
+        self.isPasswordValid = passwordValidator.validate(password)
+    }
+}
+
+// [WPB-15571] Add accessibility strings to the mask / unmask buttons
+public struct PasswordField: View {
+    @FocusState private var isFocused: Bool
+    @ObservedObject private var viewModel: PasswordFieldViewModel
+
+    private let placeholder: String
+    private let title: String
+
+    public init(
+        viewModel: PasswordFieldViewModel,
+        placeholder: String,
+        title: String
+    ) {
+        self.viewModel = viewModel
+        self.placeholder = placeholder
+        self.title = title
+    }
+
+    public var body: some View {
+        VStack(alignment: .leading, spacing: 8) {
+            Text(title)
+                .font(.subheadline)
+                .foregroundColor(calculatedColor)
+
+            ZStack {
+                if viewModel.isPasswordVisible {
+                    TextField(placeholder, text: $viewModel.password)
+                        .wireTextStyle(.body1)
+                        .textFieldStyle(RoundedBorderTextFieldStyle())
+                        .focused($isFocused)
+                } else {
+                    SecureField(placeholder, text: $viewModel.password)
+                        .textFieldStyle(RoundedBorderTextFieldStyle())
+                        .focused($isFocused)
+                }
+                HStack {
+                    Spacer()
+                    Button(action: {
+                        viewModel.isPasswordVisible.toggle()
+                    }, label: {
+                        Image(systemName: viewModel.isPasswordVisible ? "eye" : "eye.slash")
+                            .foregroundColor(.gray)
+                    })
+                    .padding(.trailing, 10)
+                }
+            }
+            .overlay(
+                RoundedRectangle(cornerRadius: 5)
+                    .stroke(
+                        calculatedColor,
+                        lineWidth: viewModel.password.isEmpty ? 0 : 1
+                    )
+            )
+
+            if let passwordRules = viewModel.passwordValidator.localizedRulesDescription,
+               viewModel.arePasswordRulesVisible {
+                Text(passwordRules)
+                    .font(.caption)
+                    .foregroundColor(calculatedColor)
+            }
+        }
+        .padding(.horizontal)
+        .onChange(of: viewModel.password, perform: { newPassword in
+            viewModel.isPasswordValid = viewModel.passwordValidator.validate(newPassword)
+        })
```

Comment:
what about moving it to viewModel:

viewModel.validate(newPassword)

```suggestion
        .onChange(of: viewModel.password, perform: viewModel.validatePassword)
```

Reply:
Different password fields have different password validation rules, and a different string used to describe them.
I'd rather have a single ViewModel class and a handful of PasswordValidator conforming classes reused across the codebase.

Reply:
Yes, as we discussed with @El-Fitz , we can reuse this field for the export backup screen. For example, the password is optional for backup, so the validation will be different.